src/components/Search.tsx

```diff
@@ -75,16 +85,56 @@ function Search({query, policyIDs}: SearchProps) {
         return null;
     }
 
+    const toggleListItem = (listItem: TransactionListItemType | ReportListItemType) => {
+        // @TODO: Selecting checkboxes will be handled in a separate PR
+        if (isReportListItemType(listItem)) {
+            Log.info(listItem?.reportID?.toString() ?? '');
+            return;
+        }
+
+        Log.info(listItem.transactionID);
+    };
+
     const ListItem = SearchUtils.getListItem(type);
-    const data = SearchUtils.getSections(searchResults?.data ?? {}, type);
+
+    const data = SearchUtils.getSections(
+        {
+            ...searchResults?.data,
+            // @TODO: Remove this comment when report items are returned from the API, uncomment it to test whether ReportListItem is displayed properly.
+            // report_0: {
+            //     reportID: 0,
+            //     reportName: 'Alice’s Apples owes $110.00',
+            //     total: 1000,
+            //     currency: 'USD',
+            //     action: 'pay',
+            // },
+        } ?? {},
+        type,
+    );
+
+    const toggleAllItems = () => {
+        if (selectedItems.length === data.length) {
+            setSelectedItems([]);
+        } else {
+            setSelectedItems([...data]);
+        }
+    };
 
     return (
         <SelectionList
+            canSelectMultiple={isLargeScreenWidth}
+            onSelectAll={toggleAllItems}
+            onCheckboxPress={toggleListItem}
             customListHeader={<SearchTableHeader data={searchResults?.data} />}
             ListItem={ListItem}
             sections={[{data, isDisabled: false}]}
             onSelectRow={(item) => {
-                openReport(item.transactionThreadReportID);
+                if (isReportListItemType(item)) {
+                    openReport(item.reportID);
+                    return;
+                }
```

Comment:
if you would prefer a bit different data flow and no `return;` then:

```suggestion
const reportID = isReportListItemType(item) ? item.reportID : item.transactionID;

openReport(reportID)
```